sp/src/game/server/hl2/npc_citizen17.cpp

```diff
@@ -1443,29 +1451,45 @@ bool CNPC_Citizen::GiveBackupWeapon( CBaseCombatWeapon * pWeapon, CBaseEntity *
 	// Don't give any more backup weapons!
 	m_bUsedBackupWeapon = true;
 
-	// Is this weapon already a side arm?
-	if ( pWeapon != NULL && pWeapon->ClassMatches( "weapon_smg1" ) || pWeapon->ClassMatches( "weapon_smg2" ) || pWeapon->ClassMatches( "weapon_pistol" ) )
+	if ( pWeapon != NULL )
 	{
-		// Very lucky citizens get crowbars as backup
-		if ( random->RandomInt( 1, 6 ) == 1 )
+		// Is this a melee weapon?
+		if ( pWeapon->IsMeleeWeapon() || pWeapon->ClassMatches( "weapon_crowbar" ) )
 		{
-			CBaseCombatWeapon * pCrowbar = GiveWeaponHolstered( AllocPooledString( "weapon_crowbar" ) );
-			pCrowbar->AddSpawnFlags( SF_WEAPON_NO_PLAYER_PICKUP );
-			pCrowbar->SetName( AllocPooledString( "worthless" ) ); // Bad Cop will say the crowbar pickup line upon interacting with this
-
-			return true;
+			// No more backups
+			return false;
 		}
+		// Is this weapon already a side arm?
+		else if ( pWeapon->ClassMatches( "weapon_smg1" ) || pWeapon->ClassMatches( "weapon_smg2" ) || pWeapon->ClassMatches( "weapon_pistol" ) )
```

Comment:
There's a few problems here - under these changes, shouldn't only pistols count as side arms? Either way, it will need to check if the gun is a handgun rather than a classname check against weapon_pistol

Reply:
The code here checks for both SMGs and pistols because other weapons (i.e. AR2s, RPGs, crossbows, etc.) are supposed to let citizens pull out a SMG1 or SMG2 instead, which is further below.

I will add a check for whether the gun is a handgun instead of just a HL2 pistol though